src/transforms/log_to_metric.rs

```diff
@@ -2003,4 +2014,42 @@ mod tests {
             .with_timestamp(Some(ts()))
         );
     }
+
+    #[tokio::test]
+    async fn transform_all_metrics_optional_namespace() {
```

Comment:
Added just one test with the `counter` metric kind. 

If you find that it would be useful to replicate this test but with the other metric kinds, tell me